lib/_http_server.js

```diff
@@ -520,11 +519,17 @@ function socketOnData(server, socket, parser, state, d) {
   onParserExecuteCommon(server, socket, parser, state, ret, d);
 }
 
+function onParserMessageBegin(parser) {
+  // We are starting to wait for the headers.
+  parser.parsingHeadersStart = nowDate();
```

Comment:
So when the parsing starts we capture when it started.

Reply:
I guess it's out of scope for this PR, but why are we depending on `Date.now()` for these things instead of the HR timer? What if a big NTP-sync shift happens during intervals like this? It could have a profound impact on how things run.